powerscale/provider/snapshot_schedule_datasource_test.go

```diff
@@ -0,0 +1,125 @@
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// filter read testing
+			{
+				Config: ProviderConfig + SnapshotScheduleDataSourceConfig,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(snapshotTerraformName, "schedules.0.duration", "604800"),
+					resource.TestCheckResourceAttr(snapshotTerraformName, "schedules.0.name", "Snapshot schedule 370395356"),
+					resource.TestCheckResourceAttr(snapshotTerraformName, "schedules.0.alias", "tfacc_Snapshot schedule 370395356_Alias"),
+					resource.TestCheckResourceAttr(snapshotTerraformName, "schedules.0.path", "/ifs/tfacc_test_dirNew"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotScheduleDataSourceAll(t *testing.T) {
+	var ssTerraformName = "data.powerscale_snapshot_schedule.all"
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// read all testing
+			{
+				Config: ProviderConfig + SnapshotScheduleAllDataSourceConfig,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet(ssTerraformName, "schedules.#"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotScheduleDataSourceGetErr(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					FunctionMocker = Mock(helper.ListSnapshotSchedules).Return(nil, fmt.Errorf("mock error")).Build()
+				},
+				Config:      ProviderConfig + SnapshotScheduleAllDataSourceConfig,
+				ExpectError: regexp.MustCompile(`.*mock error*.`),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotScheduleDataSourceGetErrCopyAll(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					FunctionMocker = Mock(helper.CopyFields).Return(fmt.Errorf("mock error")).Build()
+				},
+				Config:      ProviderConfig + SnapshotScheduleAllDataSourceConfig,
+				ExpectError: regexp.MustCompile(`.*mock error*.`),
+			},
+		},
+	})
+}
+
+func TestAccSnapshotScheduleDataSourceGetErrCopy(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					FunctionMocker = Mock(helper.CopyFields).Return(fmt.Errorf("mock error")).Build()
+				},
+				Config:      ProviderConfig + SnapshotScheduleDataSourceConfig,
+				ExpectError: regexp.MustCompile(`.*mock error*.`),
+			},
+		},
+	})
+}
+
+var SnapshotScheduleDataSourceConfig = `
+data "powerscale_snapshot_schedule" "test" {
+  filter {
```

Comment:
@nikitajoshi1 can you try with other parameters also like dir, sort etc.?

Reply:
Added test for other filters.